Open Judge System/Workers/OJS.Workers.Executors/RestrictedProcessExecutor.cs

```diff
@@ -160,5 +160,140 @@ public ProcessExecutionResult Execute(string fileName, string inputData, int tim
 
             return result;
         }
+
+        public ProcessExecutionResult ExecuteJavaProcess(string fileName, string inputData, int timeLimit, int memoryLimit, string workingDirectory, IEnumerable<string> executionArguments = null)
+        {
+            var result = new ProcessExecutionResult { Type = ProcessExecutionResultType.Success };
+
+            using (var restrictedProcess = new RestrictedProcess(fileName, workingDirectory, executionArguments, Math.Max(4096, (inputData.Length * 2) + 4)))
+            {
+                // Write to standard input using another thread
+                restrictedProcess.StandardInput.WriteLineAsync(inputData).ContinueWith(
+                    delegate
+                    {
+                        // ReSharper disable once AccessToDisposedClosure
+                        if (!restrictedProcess.IsDisposed)
+                        {
+                            // ReSharper disable once AccessToDisposedClosure
+                            restrictedProcess.StandardInput.FlushAsync().ContinueWith(
+                                delegate
+                                {
+                                    restrictedProcess.StandardInput.Close();
+                                });
+                        }
+                    });
+
+                // Read standard output using another thread to prevent process locking (waiting us to empty the output buffer)
+                var processOutputTask = restrictedProcess.StandardOutput.ReadToEndAsync().ContinueWith(
+                    x =>
+                    {
+                        result.ReceivedOutput = x.Result;
+                    });
+
+                // Read standard error using another thread
+                var errorOutputTask = restrictedProcess.StandardError.ReadToEndAsync().ContinueWith(
+                    x =>
+                    {
+                        result.ErrorOutput = x.Result;
+                    });
+
+                // Read memory consumption every few milliseconds to determine the peak memory usage of the process
+                const int TimeIntervalBetweenTwoMemoryConsumptionRequests = 45;
```

Comment:
The constant is the same as the one in the other method, maybe it can be extracted.